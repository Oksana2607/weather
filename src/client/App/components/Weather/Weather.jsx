import React from 'react';

import {
  Table,
  TableCaption,
  TableBody,
  TableRow,
  TableH,
  TableD,
  TableHr
} from './styledComponents';

const Weather = (props) => {
  const {
    weather,
  } = props;

  const hasWeather = Object.keys(weather).length > 0;

  return (
    <>
      <Table>
        <TableCaption>RIGHT NOW</TableCaption>
        { hasWeather && (
        <TableBody>
          <TableRow>
            <TableH>Wind</TableH>
            <TableD>{weather.wind.speed}</TableD>
          </TableRow>
          <TableHr />
          <TableRow>
            <TableH>Humidity</TableH>
            <TableD><span>{weather.main.humidity}</span></TableD>
          </TableRow>
          <TableHr />
          <TableRow>
            <TableH>Pressure</TableH>
            <TableD>{weather.main.pressure}</TableD>
          </TableRow>
          <TableHr />
          <TableRow>
            <TableH>Visibility</TableH>
            <TableD>{weather.visibility}</TableD>
          </TableRow>
          <TableHr />
          <TableRow>
            <TableH>Min temp</TableH>
            <TableD>
              {weather.main.temp_min}
              <sup>°</sup>
            </TableD>
          </TableRow>
          <TableHr />
          <TableRow>
            <TableH>Max temp</TableH>
            <TableD>
              {weather.main.temp_max}
              <sup>°</sup>
            </TableD>
          </TableRow>
        </TableBody>
        ) }
      </Table>
    </>
  );
};

export default Weather;
