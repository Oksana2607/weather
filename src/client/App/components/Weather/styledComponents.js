import styled from 'styled-components';

export const Table = styled.table`
    font-weight: bold;
`;

export const TableCaption = styled.caption`
    padding: 0.3em;
    text-align: left;
`;

export const TableBody = styled.tbody``;

export const TableRow = styled.tr`
    color: rgb(57, 57, 57);
    vertical-align: middle;
    border-bottom: 1px solid rgb(57, 57, 57);
`;

export const TableH = styled.th`
    text-align: left;
    display: block;
    font-size: 16px;
    font-weight: 600;
    padding: 0.3em 0.8em;
`;

export const TableD = styled.td`
    display: block;
    font-size: 14px;
    line-height: 18px;
    color: rgb(57, 57, 57);
    padding: 0.1em 0.8em;
`;

export const TableHr = styled.hr`
    width: 130px;
`;

