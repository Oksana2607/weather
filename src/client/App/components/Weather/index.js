import { connect } from 'react-redux';
import * as actions from '../../../actions';
import * as selectors from '../../../selectors/selectors';
import Weather from './Weather';

const mapStateToProps = state => ({
  weather: selectors.getWeather(state),

});

const mapDispatchToProps = dispatch => ({

});

export default connect(mapStateToProps, mapDispatchToProps)(Weather);
