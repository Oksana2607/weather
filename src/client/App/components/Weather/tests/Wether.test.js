import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Weather from '../Weather';

configure({ adapter: new Adapter() });

describe('CitiesTable snapshot', () => {
  it('should render correctly', () => {
    const mockProps = {
      weather: {}
    };

    const wrapper = shallow(
      <Weather {...mockProps} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
