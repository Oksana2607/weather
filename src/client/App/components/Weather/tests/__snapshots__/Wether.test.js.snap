// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesTable snapshot should render correctly 1`] = `ShallowWrapper {}`;
